docs/source/reproducers/03-zuul.md

```diff
@@ -137,6 +145,72 @@ $ tail -f ~/ansible.log
 
 Note that `tmux` is installed by default, and may be handy in such a case ;).
 
+#### Zuul_inventory.yml
+That file is a generated inventory, allowing ansible to know about the compute(s) and CRC nodes.
+
+It exposes each node with the connection information (user and IP), and also created groups in
+order to access, for instance, all of the computes. Those groups mimic the same behaviour we
+have in Zuul.
+
+### Ansible tags of interest
+You may want to re-run some parts of the reproducer only, and avoid re-running the whole
+virtual machine creation by using `--skip-tags TAG`.
+
+In order to do so, the reproducer exposes two tags:
+* bootstrap_layout
+* bootstrap
+
+#### Tag: bootstrap_layout
+This covers everything related to "create and manage virtual machines". This is probably the
+most useful tag to skip once you want to iterate on a deployed layout, since it will
+ensure repositories are up-to-date, but also re-generate the needed bits to run the CI job.
+
+Note: you **must** have an already deployed layout before skipping this tag.
+
+##### Usage
+```Bash
+# Reproduce first iteration of your PR that failed on Zuul
+$ ansible-playbook -i custom/inventory.yml reproducer.yml \
+    -e cifmw_target_host=builder1 \
+    -e @scenarios/reproducers/3-nodes.yml \
+    -e cifmw_job_uri="YOUR_URI"
+
+# Iterate with the second iteration of your PR that failed on Zuul
+$ ansible-playbook -i custom/inventory.yml reproducer.yml \
+    -e cifmw_target_host=builder1 \
+    -e @scenarios/reproducers/3-nodes.yml \
+    -e cifmw_job_uri="YOUR_UPDATE_URI" \
+    --skip-tags bootstrap_layout
+```
+
+#### Tag: bootstrap
+This covers everything from the creation of the virtual layout (networks and machines) to
```

Comment:
Hmm... I'm confused by lines 164 and 187.
IIUC `bootstrap_layout` should give me working environment for VMs: setup of network, create VM, start VM. `bootstrap` on the other hand should download me repositories, zuul config (data), etc.
However line 187 mentions: "everything from [..] virtual layout to [..] plays".
Should it be: "This covers everything *after* virtual layout"?

So:
`bootstrap_layout` + `bootstrap` == first config
`bootstrap` == subsequent runs

Reply:
if you check the actual tags associated to the tasks, you will see a fair overlap of both `bootstrap` and `bootstrap_layout`. The `bootstrap` is a pre-existing tag, reused for some of the templates, repositories and related tasks in this context. The `bootstrap_layout` has been introduced in order to select only the VM parts.
You may see that `bootstrap` *also* covers some of the VM parts, in addition to the repositories/templates management.

Also keep in mind those tags are supposed to be *skipped*, not targeted using `--tags` - they represent tasks to exclude from a run.